x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -666,15 +668,22 @@ export class AlertsClient {
         authorizedFeatures.add(ruleType.producer);
       }
 
-      const toReturn = Array.from(authorizedFeatures).flatMap((feature) => {
-        if (featureIds.includes(feature) && isValidFeatureId(feature)) {
-          if (feature === 'siem') {
-            return `${mapConsumerToIndexName[feature]}-${this.spaceId}`;
-          } else {
-            return `${mapConsumerToIndexName[feature]}`;
-          }
+      const validAuthorizedFeatures = Array.from(authorizedFeatures).filter(
+        (feature): feature is ValidFeatureId =>
+          featureIds.includes(feature) && isValidFeatureId(feature)
+      );
+
+      const toReturn = validAuthorizedFeatures.flatMap((feature) => {
+        const indices = this.ruleDataService.findIndicesByFeature(feature, Dataset.alerts);
+        if (feature === 'siem') {
+          // TODO: Remove space id from the index name and make sure the app works well.
+          // We should not include space id into the index name, because a
+          // namespace goes into the index name, and it's user-defined in general.
+          // The user can set a custom namespace per rule instance which could be != space id.
```

Comment:
namespaces can have whitespace chars in them, correct? That could be a reason to use space id over namespace.

Reply:
We had a chat with @dhurley14 on this comment. For visibility I will post a summary here.

My point was: namespaces, according to the design document, are supposed to be user-defined. But in Security they will be equal to the space id by default. But still, the user will be able to set a custom namespace for a single rule instance.
- if we include a space id in the index name when reading data, we will omit the alerts generated by this rule instance
- namespaces should not include whitespace characters I guess, but they can contain dashes; basically they can contain any characters a space id can
- when reading from RAC indices, we should omit the namespace part from the queries to ES

Devin's point is:

> that all makes sense except for the last bullet point:
> > when reading from RAC indices, we should omit the namespace part from the queries to ES
>
> We are doing that as part of the authorization filter. This requirement came early on from Brandon I think so that would need to be discussed further.
> I'm fine with whichever decision is made going forward (sounds like it'll be namespaces > space ids) but I just want to make sure that change is also reflected in places we use space id for authorization in the alerts as data client.

Im not sure what should be done in this PR regarding that, to be honest. Maybe I just don’t understand why this function (`getAuthorizedAlertsIndices`) needs to include space id to the index name. I don’t know how the HTTP endpoint that calls this function is actually used in the app. Where the index names in the response are being propagated to, etc. Maybe I'm just missing something.

I will remove this TODO comment to avoid confusion (especially if it's misleading) and try to figure out the usage of space id in this function later.